src/hypertable_restrict_info.c

```diff
@@ -686,16 +699,21 @@ ts_hypertable_restrict_info_get_chunks(HypertableRestrictInfo *hri, Hypertable *
 		}
 
 		/*
-		 * Always include the OSM chunk if we have one. It has some virtual
-		 * dimension slices (at the moment, (+inf, +inf) slice for time, but it
-		 * used to be different and might change again.) So sometimes it will
-		 * match and sometimes it won't, so we have to check if it's already
-		 * there not to add a duplicate.
+		 * Always include the OSM chunk if we have one and OSM reads are
+		 * enabled. It has some virtual dimension slices (at the moment,
+		 * (+inf, +inf) slice for time, but it used to be different and might
+		 * change again.) So sometimes it will match and sometimes it won't,
+		 * so we have to check if it's already there not to add a duplicate.
+		 * Similarly if OSM reads are disabled then we exclude the OSM chunk.
 		 */
 		int32 osm_chunk_id = ts_chunk_get_osm_chunk_id(ht->fd.id);
-		if (osm_chunk_id != 0 && !list_member_int(chunk_ids, osm_chunk_id))
+
+		if (osm_chunk_id != 0)
 		{
-			chunk_ids = lappend_int(chunk_ids, osm_chunk_id);
+			chunk_ids = !ts_guc_enable_osm_reads ? list_delete_int(chunk_ids, osm_chunk_id) :
```

Comment:
The nested expression is difficult to understand. Is it possible to write it down in a more verbose way?

Reply:
I converted it into a regular if/else statement